edge-runtime/lib/response.ts

```diff
@@ -187,26 +181,12 @@ export const buildResponse = async ({
     // respect trailing slash rules to prevent 308s
     rewriteUrl.pathname = normalizeTrailingSlash(rewriteUrl.pathname, nextConfig?.trailingSlash)
 
-    const target = normalizeLocalizedTarget({ target: rewriteUrl.toString(), request, nextConfig })
-    if (target === request.url) {
-      logger.withFields({ rewrite_url: rewrite }).debug('Rewrite url is same as original url')
-      return
-    }
```

Comment:
is this check not needed anymore?

Reply:
I did some digging after your comment above about the `normalizeLocalizedTarget` params because I realised that I don't think we actually need `normalizeLocalizedTarget` at all. All it does is force the locale to be whatever was specified in the target path or the current locale. However, that's the not the behaviour we want - Next.js doesn't default to the current locale if you don't specify a locale in the rewrite/redirect path, it just redirects to whatever the target URL is.

Reply:
It seems to be the behaviour we need (from my manual testing) and all our tests pass. I'm going to run the full e2e test suite on it and double check.

Reply:
I was thinking about https://github.com/netlify/next-runtime-minimal/pull/193 and the check was added to avoid looping apparently? Do we handle this in other way now / looping doesn't happen anymore?

Reply:
Ah, I see what you mean. No it's fine because we already check that condition on [Line 135](https://github.com/netlify/next-runtime/pull/2548/files/a7620fdd9009f25fb7820dd43828a30aa22ff434..da8215f0d9d05aef90526e957ced6447f8567c0c#diff-4f9f8de63d5504fcd1a98cf04a47b85e6759a68279a524e2713058bf4564a99eR135). The second check was because previously we may have mutated the target URL in `normalizeLocalizedTarget`, but we no longer use that.